tools/approval-service/internal/approvalservice/sources/srv_test.go

```diff
@@ -0,0 +1,53 @@
+package sources
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestServer(t *testing.T) {
+	tests := map[string]struct {
+		opts        []ServerOpt
+		expectError bool
+	}{
+		"valid options": {
+			opts: []ServerOpt{
+				WithAddress(":8080"),
+				WithHandler("/path", okHandler),
+			},
+			expectError: false,
+		},
+		"invalid address": {
+			opts: []ServerOpt{
+				WithAddress(""),
+			},
+			expectError: true,
+		},
+		"path conflicts": {
+			opts: []ServerOpt{
+				WithAddress(":8080"),
+				WithHandler("/path", okHandler),
+				WithHandler("/path", okHandler),
+			},
+			expectError: true,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			_, err := NewServer(tt.opts...)
+			if tt.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
```

Comment:
The tests do not test that the handlers can be called. Is that missing here or will it be added shortly?